woof-code/rootfs-packages/firewall_ng/usr/sbin/firewall_ng

```diff
@@ -1448,7 +1448,7 @@ else
 	exit
 fi
 
-[ "$1" = "enable" ] && exit
```

Comment:
This PR breaks this line: https://github.com/puppylinux-woof-CE/woof-CE/blob/28f7424eef841fb3591b9c989c7d5bfd25153128/woof-distro/x86_64/debian/bullseye64/_00build.conf#L132

How about `[ "$1" = "enable" -o "$MAIN" = "false" ] && exit` instead?

Reply:
Hmm, let me study this further....

Reply:
Well...I'm not fluent enough with this build system (or github for that matter) to understand what is going on here. Could it have been from the missing space in line 1451? I have corrected that now in my patch1 fork.

Reply:
In regards to using the `[ "$1" = "enable" -o "$MAIN" = "false" ] && exit` statement. It would allow the "enable" test to exit before restarting the firewall. I am thinking we would only want to exit on a "false" test on this line and allow "enable" to continue further down the source code. No?

Reply:
> I am thinking we would only want to exit on a "false" test on this line and allow "enable" to continue further down the source code. No?

One line with two condition with OR between them or two separate conditions in two lines - doesn't matter. But don't break support for `firewall_ng enable`, things use it.

Reply:
OK, understood.  `[ "$1" = "enable" -a "$MAIN" = "false" ] && exit` still keeps Cancel and Apply buttons functionality intact. I will modify my patch1 source. See if that builds?

Reply:
I realize I am using a AND instead of a OR here. Using a OR here breaks code continuation to complete file save and firewall restart.

Reply:
But that's what you want to do. `firewall_ng enable` is used to enable the firewall without starting it (hence, must exit at this point) and if `MAIN=false`, the user chose to disable the firewall (hence, must exit at this point).

Reply:
I'm not quite sure I understand. So using a AND here is acceptable? Otherwise like I said, using a OR here breaks firewall restart.

Reply:
With AND, the condition is _"asked to exit without starting the firewall" AND "the firewall is disabled"_ - this condition is always false when you start firewall_ng via the menu entry (because the left part is false).

With OR, the condition is _"asked to exit without starting the firewall" OR "the firewall is disabled"_, which makes sense.